### DISCUSSION TITLE
Project maintained??

### DISCUSSION THREAD

**Discussion by pada57** [rasmus/EventFlow#974] [General]
Hi
As looking for good base for starting DDD base project with event driven approach, this project would be interesting if currently maintained ?

What is foreseen roadmap if this is the case?

**Comment by rasmus** [comment]
It is maintained. Though a little slow these days On 3 Apr 2023, at 21.00, pada57 ***@***.***> wrote:﻿
Hi
As looking for good base for starting DDD base project with event driven approach, this project would be interesting if currently maintained ?
What is foreseen roadmap if this is the case?

**Reply by pada57** [reply]
Is there a roadmap? I see last stable release 2 years ago isn't it?

**Reply by rasmus** [reply]
There is not current roadmap. Last release was a 1.0 pre-release in November. Currently waiting on some copyrights cleanup with Schibsted. But I just started on the prep work for the actual 1.0 release.

**Reply by gius** [reply]
Hi, do you have any update on this topic? Do you plan to release v1?

**Reply by rasmus** [reply]
I have had quite a lot on my hands since last update. I really want to get back on this project and hope to do so soon.

One of the problems that I have been facing is the copyrights noted in this PR https://github.com/eventflow/EventFlow/pull/968. While I did have everything almost ready, the manager handling the case went on maternity leave. She absolutely deserves it and I fully support it, but it did put everything on hold. She's back and I hope to pick this up again very soon.

I have also been hunting for a few people in my personal circle to help support the project to have it less vulnerable to one person being out for a while.

So while there haven't happened much on the project, something is happening "off line".

I hope to clear everything up. If you want to support getting v1 out the door, let me know and we can coordinate it. I'm more than open to get PRs in.

**Reply by rasmus** [reply]
The legal stuff just went through! Picking up again and preparing another alpha release.

Hopefully I can get more done and finally release 1.0 properly.

**Comment by quezlatch** [comment]
Hi,

glad to hear things are coming along. do you have an idea for release 1.0 timeline? i'm happy to get involved if that would help

regards,

**Comment by rasmus** [comment]
Hi @quezlatch 

Regarding 1.0, I have been thinking of getting the projects that are currently migrated released and then do the others along the way in future 1.0 releases. Not ideal, but since many of the core packages are done, it will provide value to a many. There's still a few things I would like to have done before 1.0

- Migrate documentation to mkdocs and hosted on GitHub pages. Its mostly done to make it a bit simpler, but also be ensure that no tracking cookies are injected and ads are displayed.
- Provide native System.Text.Json support. Its something I have been struggling with back and forth, but removing Newtonsoft.Json from the base types seems too big of change that can really break a lot for many (on top of many other changes for 1.0). But adding the support of System.Text.Json as and additional extension method would help people switch if they want. The only down side is that they will get a dead DLL as part of the build

As for motivation. For me it very much goes up and down and its has certainly been in a down period. Doing open-source is not something I would recommend anyone. Its a lot of responsibility and very little (if any) gains, thus its hard to "find" the time to do more work in the evenings, especially since I have a family that also demands my time and for the little "me" time I have, Baldur's Gate 3 is such a temptation. I do however have a deep respect for the developers using what I have created and really want to ensure the project moves forward.

If you (or anyone) want to help, then perfect! Do small PRs that do specific changes (unless its the documentation, that can be hacked at), it will make it easier to review and get the changes in. It also makes it easier to merge changes from v0 as I haven't completely written that off just yet.

If anyone wants to be more involved in the project, I'll all for it. Although with the recent [xz backdoor](https://www.wired.com/story/xz-backdoor-everything-you-need-to-know/), I won't be handing any "keys to the kingdom", which I guess anyone using EventFlow will appreciate. 

❤️ rasmus

**Reply by quezlatch** [reply]
thank you for your very thoughtful reply. i certainly understand the difficulties of finding time! i would be keen to help (and by necessity they would be small PRs 🙂  ). I have a particular interest in using azure storage for persistence, but if there are more general things that need doing, I'm happy to be involved

**Reply by Tsvetkraft** [reply]
Dear @rasmus , i also want to help with this great project. Is there some list of tasks or something?

**Reply by rasmus** [reply]
There certainly is. I would really like to have the documentation cleaned up before I release 1.0. It's a rather big task and I struggle to get it done.

Then there's the migration of the projects that isn't migrated to 1.0. Not a required thing as developers can stay on v0 until their dependencies are done.

Then there's v0 itself. I had originally thought of abandoning it. But it seems that some are still interested in using it as they have some legacy systems. Here the tests needs to be migrated to GHA. I think 1.0 stuff is more important as that is for Microsoft supported .NET versions.


Thanks.

**Comment by rasmus** [comment]
Both v0 and v1 are now getting some proper loving
